apps/src/templates/progress/GoogleClassroomShareButton.jsx

```diff
@@ -0,0 +1,103 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Button from '../Button';
+import i18n from '@cdo/locale';
+
+// https://developers.google.com/classroom/brand
+const styles = {
+  label: {
+    paddingLeft: 16,
+    fontFamily: 'Roboto',
+    fontWeight: 700
+  },
+  container: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'start'
+  }
+};
+class GoogleClassroomShareButton extends React.Component {
+  static propTypes = {
+    buttonId: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+    itemtype: PropTypes.string.isRequired,
+    title: PropTypes.string,
+    height: PropTypes.number,
+    courseid: PropTypes.number
+  };
+
+  constructor(props) {
+    super(props);
+    this.loadApi = this.loadApi.bind(this);
+    this.waitForGapi = this.waitForGapi.bind(this);
+    this.renderButton = this.renderButton.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.gapiReady()) {
+      this.renderButton();
+    } else {
+      this.loadApi();
+    }
+  }
+
+  /* The Google Classroom Share Button is only available through their js api,
+   * so we have to add it to the page on load. Each instance of this
+   * component will wait until the api is ready before rendering the button,
+   * but only the first will kick of the loading process.
+   */
+  loadApi() {
+    if (!document.getElementById('gapi')) {
+      window.___gcfg = {
+        parsetags: 'explicit'
+      };
+
+      const gapi = document.createElement('script');
+      gapi.src = 'https://apis.google.com/js/platform.js';
+      gapi.id = 'gapi';
+      gapi.addEventListener('load', this.waitForGapi);
+      document.body.appendChild(gapi);
+    } else {
+      this.waitForGapi();
+    }
+  }
+
+  waitForGapi() {
+    if (this.gapiReady()) {
+      this.renderButton();
+    } else {
+      setTimeout(() => {
```

Comment:
Not sure whether this is an issue: If apis.google.com is down or unreachable, will this keep spinning forever? If so, should we add a timeout?

Reply:
you thinking like set the start time on initial load then stop trying after some given interval? it's a pretty cheap operation so that could be overkill, but i'm happy to add it. 

another option would be to use exponential backoff (e.g. multiply the timeout interval by 2 each time) -- that might be a more robust than setting a specific "stop trying" interval in the case that it's just taking a really long time to load.

Reply:
Hmm. I don't know if I have a specific preference. Just thinking it might be nice to communicate the state to users. Definitely not a hard requirement.

Reply:
@clareconstantine what do you think?

at the very least, it might be good to not render this component at all until the api is ready. right now we render the the label immediately without leaving space for the button, so when the api loads the label gets moved over, which looks a little wonky.

another option would be to render a loading spinner as a placeholder for the button, which we could replace with a "failed to load" icon on timeout if we take that approach. or it could just keep spinning indefinitely if we go with exponential backoff.

Reply:
per a [conversation with Marina](https://codedotorg.slack.com/archives/CA3KCSGTD/p1600112325161500) in slack, i'm going to not render anything until the api loads and add a timeout. i will also add metrics for api load time and failed loads in #36728

Reply:
I would lean towards not showing anything until the api loads - if we get a zendesk ticket of someone not seeing the button, they will still have an error in the console, and it just seems cleanest. Not sure how helpful this would be since most users might not know to check the console, but we could put a link to the sites to whitelist in a comment in the console if the api doesn't load, in case they check there.

Reply:
oh just saw your comment :) sounds good!